saleor/graphql/order/mutations/fulfillments.py

```diff
@@ -531,16 +531,19 @@ class Meta:
 
     @classmethod
     def clean_order_payment(cls, payment, cleaned_input):
-        if not payment or not payment.can_refund():
-            raise ValidationError(
-                {
-                    "order": ValidationError(
-                        "Order cannot be refunded.",
-                        code=OrderErrorCode.CANNOT_REFUND.value,
-                    )
-                }
-            )
-        cleaned_input["payment"] = payment
+        if payment:
+            if not payment.gateway or payment.can_refund():
```

Comment:
We can join it into one condition:
```suggestion
        if payment and (not payment.gateway or payment.can_refund()):
```